### ISSUE TITLE
mpirun and prterun do not return when run without a target

### ISSUE THREAD

**Issue by david-edwards-arm** [openpmix/prrte#1352]
### What version of the PMIx Reference Server are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)
Open MPI 5.0.0rc6

##### What version of PMIx are you using? (e.g., v1.2.5, v2.0.3, v2.1.0, git branch name and hash, etc.)
Open MPI 5.0.0rc6

### Please describe the system on which you are running

* Operating system/version: RHEL 7
* Computer hardware: x86_64

-----------------------------

## Details of the problem

Running mpirun or prterun either with no arguments or with a single '-v' (or '--verbose') argument results in the command not returning, whereas expected behaviour would be that the command completes and an error is provided indicating the target is missing.
For example Open MPI 4.1.0 mpirun outputs `No executable was specified on the mpirun command line.` in this scenario.
The issue in 5.0.x likely relates to mpirun execv-ing prterun and prterun being a symlink to prte.
The reason this causes an issue is that help and version arguments are not uniform across different MPI implementations, and speculatively running launchers with different known such arguments (including '-v') can help with auto-detecting which MPI is loaded into the environment.

**Comment by rhc54** [comment]
The `prte` command underlying all that interprets the lack of an executable to mean that it is being invoked as a DVM, which is why it doesn't return. I'll have to add some logic to allow the proxy detection to dictate that behavior.

**Comment by rhc54** [comment]
Hmmm...I am unable to replicate this:
```text
$ prterun -v
--------------------------------------------------------------------------
No executable was specified on the prterun command line.

Aborting.
--------------------------------------------------------------------------

$ prterun
--------------------------------------------------------------------------
No executable was specified on the prterun command line.

Aborting.
--------------------------------------------------------------------------
$ 
```
Is it possible there is something wrong in the OMPI-side integration? You might check with them.

**Comment by naughtont3** [comment]
Looks like this is still an issue when I did check of latest ompi-5.0.x branch (after v5.0.0rc6).  So I can take a look/lend a hand.

**Comment by rhc54** [comment]
Compare the code in `prte.c` between master and v2.1 branches where we check to see if an app was provided on the cmd line (starts at line 471 on master). At first glance, it looks correct to me - if so, then the problem must be in the cmd parser (src/util/cmd_line.c in v2.1) as it must think it has an app.

**Comment by naughtont3** [comment]
Testing with latest prte/pmix in ompi-v5.0.x branch (openpmix@7ddb00e, prrte@407e8d5).  It appears there is something getting messed up with the libevent/show-help setup.  The code is detecting there are no apps and calls the show_help for `run:executable-not-specified`, but that just sits and spins after showing a "warn" message,
```
[warn] event_active:  event has no event_base set.
```

So I guess need to look earlier on in the setup to see if something got confused when initializing the IOF/output stuff.

**Comment by naughtont3** [comment]
i have to run for today, but adding quick note.  need to look closer at the show_help stuff.  The version of prte in ompi-v5.0.x is still based on the `prte/src/util/show_help.c`, instead of using the pmix show_help like in prte master.

**Comment by rhc54** [comment]
I think I'm concluding that we should bring the PMIx/PRRTE utility changes to the release branch. There is just too much change, and trying to backport PRRTE changes is going to become increasingly problematic. Probably the better part of valor is to just bite the bullet and do it.

Means bringing those changes to the PMIx v4.2, and probably v4.1 branches - and then updating the PRRTE side of things. I can take a look at it tomorrow.

**Comment by rhc54** [comment]
@naughtont3 I didn't immediately grok that the "show_help" logging PR hadn't been committed yet here. Try it again now that it has been added.

@awlauria backporting the show_help changes to PMIx v4.1 is probably something we should do. We'll need to see what has to be done here to let PRRTE at least run with something like v4.1.0 (since that is already released) - if that proves too burdensome, then we need to adjust the configure logic to reject those versions.

**Comment by naughtont3** [comment]
@rhc54 & @awlauria I think the latest changes for prte-v2.1 and pmix-v4.2 resolve the `prterun`  (no args) hang for show_help.  So i guess that is better, possibly minimal needed.

**Comment by rhc54** [comment]
Only thing that bothers me is that you were hanging without either patch - I thought you had one of the two sides patched and the other not. However, this is the configuration we are going to ship, so I won't worry about it.

**Comment by naughtont3** [comment]
I think the best check will be to test once the next ompi-5.0.0rc is rolled.  That way we know it is good for that release.

**Comment by david-edwards-arm** [comment]
Thanks all - I fetched the latest openpmix v4.2 (8c39d8e6) and prrte v2.1 (f75647a0) and rebuilt, however I don't see any output from `prterun -v`, `prterun --version` or `prterun --help`.  Maybe something is wrong with my build/install... I'd welcome someone else checking.
Also on this branch I can still repro some of the other recent tool issues - I'll follow up on these separately.

**Comment by rhc54** [comment]
This has been fixed with the update to PRRTE v2.1:
```text
$ prterun -v
--------------------------------------------------------------------------
No executable was specified on the prterun command line.

Aborting.
--------------------------------------------------------------------------

$ prterun --version
prterun (PMIx) 4.2.0

Report bugs to: https://github.com/openpmix/prrte

$ prterun --help
prterun (PMIx) 4.2.0

Usage: prterun [OPTION]...
Initiate an instance of the PMIx Reference RTE (PRRTE) DVM

/*****      General Options      *****/
....
```
